src/main/cli/cli.c

```diff
@@ -6693,24 +6693,33 @@ static void processCharacter(const char c)
             if (cmd < cmdTable + ARRAYLEN(cmdTable)) {
                 cmd->cliCommand(cmd->name, options);
             } else {
-                cliPrintError("input", "UNKNOWN COMMAND, TRY 'HELP'");
+                if (cliInteractive) {
+                    cliPrintError("input", "UNKNOWN COMMAND, TRY 'HELP'");
+                } else {
+                    cliPrint("ERR_CMD_NA: ");
+                    cliPrintLine(cliBuffer);
+                }
             }
             bufferIndex = 0;
         }
 
         memset(cliBuffer, 0, sizeof(cliBuffer));
 
-        // 'exit' will reset this flag, so we don't need to print prompt again
-        if (!cliMode) {
-            return;
+        // prompt if in interactive mode
```

Comment:
Not valid. You are either interactive or not. There is no ability to change the state if interactive, therefore there is no "still".

Reply:
this also prevents printing prompt after `exit`

https://github.com/betaflight/betaflight/pull/13940/files#diff-92382fe98d1b21a00115a698607460fe13ad7bdf44208bfd55c66b265268c076R6812-R6813

Reply:
It is a superfluous word and poor English. 

The flag is cliInteractive, if it is true the cli is interactive mode - there is no possible change to interactive mode in the session (before cliExit or cliReboot). Therefore given there is no state change the use of the word still is a poor choice, as it implies an ability for the state to change.

Reply:
Sorry, English isn't my first language and my grammar is poor.

The idea is: 'If last command did not leave interactive mode, print another prompt.' State may change inside this function

Reply:
Thanks for the explanation. It is possible the state will change due to the functions, exit, defaults or save, but the session has ended at this point, and the word `still` is superfluous. Better would be to have a check after command for `if (!cliMode) return;` to avoid any ambiguity.